core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+      InputMethodManager.HIDE_IMPLICIT_ONLY,
+      0
+    )
+  }
+
+  private fun saveNote(noteText: String) {
+
+    /* String content of the EditText, given by noteText, is saved into the text file given by:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * */
+    if (instance.isExternalStorageWritable) {
+      if (ContextCompat.checkSelfPermission(
+          requireContext(),
+          Manifest.permission.WRITE_EXTERNAL_STORAGE
+        ) != PackageManager.PERMISSION_GRANTED
+      ) {
+        Log.d(
+          TAG,
+          "WRITE_EXTERNAL_STORAGE permission not granted"
+        )
+        context.toast(R.string.note_save_unsuccessful, Toast.LENGTH_LONG)
+        return
+      }
+      val notesFolder = File(zimNotesDirectory)
+      var folderExists = true
+      if (!notesFolder.exists()) {
+        // Try creating folder if it doesn't exist
+        folderExists = notesFolder.mkdirs()
+      }
+      if (folderExists) {
+        val noteFile =
+          File(notesFolder.absolutePath, "$articleNoteFileName.txt")
+
+        // Save note text-file code:
+        try {
+          val fileOutputStream = FileOutputStream(noteFile)
+          fileOutputStream.write(noteText.toByteArray())
+          fileOutputStream.close()
+          context.toast(R.string.note_save_successful, Toast.LENGTH_SHORT)
+          noteEdited = false // As no unsaved changes remain
+          enableDeleteNoteMenuItem()
+        } catch (e: IOException) {
+          e.printStackTrace()
+            .also { context.toast(R.string.note_save_unsuccessful, Toast.LENGTH_LONG) }
+        }
+      } else {
+        context.toast(R.string.note_save_successful, Toast.LENGTH_LONG)
+        Log.d(TAG, "Required folder doesn't exist")
+      }
+    } else {
+      context.toast(R.string.note_save_error_storage_not_writable, Toast.LENGTH_LONG)
+    }
+  }
+
+  private fun deleteNote() {
+    val notesFolder = File(zimNotesDirectory)
+    val noteFile =
+      File(notesFolder.absolutePath, "$articleNoteFileName.txt")
+    val noteDeleted = noteFile.delete()
+    if (noteDeleted) {
+      add_note_edit_text.text.clear()
+      disableMenuItems()
+      context.toast(R.string.note_delete_successful, Toast.LENGTH_LONG)
+    } else {
+      context.toast(R.string.note_delete_unsuccessful, Toast.LENGTH_LONG)
+    }
+  }
+
+  /* String content of the note text file given at:
+   *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+   * is displayed in the EditText field (note content area)
+   */
+  private fun displayNote() {
+
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      writeNote(noteFile)
+    }
+
+    // No action in case the note file for the currently open article doesn't exist
+  }
+
+  private fun writeNote(noteFile: File) {
+    noteFileExists = true
+    val contents = StringBuilder()
+    try {
+      noteFile.bufferedReader().use { bufferReader ->
```

Comment:
`File.readText()` will probably do here

Reply:
@macgills How should I append this line with File.readText() ?
` contents.append(System.getProperty("line.separator"))`

Reply:
I don't think that line really matters, we are just taking and putting text to a file, I am not concerned with separators, they will be there or they won't be